File: app/src/main/java/com/leodeleon/popmovies/feature/MainActivity.java
package com.leodeleon.popmovies.feature;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v7.app.AppCompatActivity;
import android.view.MenuItem;
import com.leodeleon.popmovies.R;
import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.support.HasSupportFragmentInjector;
import javax.inject.Inject;

public class MainActivity extends AppCompatActivity implements HasSupportFragmentInjector {
  @Inject DispatchingAndroidInjector<Fragment> dispatchingAndroidInjector;

  private FragmentManager manager = getSupportFragmentManager();

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);
  }


  @Override
  public boolean onOptionsItemSelected(MenuItem item) {
    switch (item.getItemId()) {
      case android.R.id.home:
        onBackPressed();
        return true;
    }
    return super.onOptionsItemSelected(item);
  }

  @Override public void onBackPressed() {
    if (manager.getBackStackEntryCount() == 0) {
      super.onBackPressed();
    }else {
      manager.popBackStackImmediate();
    }
  }

  @Override public AndroidInjector<Fragment> supportFragmentInjector() {
    return dispatchingAndroidInjector;
  }



  public void addFragment(Fragment fragment) {
    manager.beginTransaction().add(R.id.container, fragment).addToBackStack(fragment.toString()).commit();
  }

}

File: app/src/main/java/com/leodeleon/popmovies/feature/MainActivity.kt
package com.leodeleon.popmovies.feature

import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.app.AppCompatActivity
import android.view.MenuItem
import androidx.navigation.NavController
import androidx.navigation.Navigation
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.NavigationUI
import com.leodeleon.popmovies.R
import com.leodeleon.popmovies.util.listen
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {
  lateinit var host: NavHostFragment
  private val controller: NavController by lazy {
    Navigation.findNavController(this, R.id.nav_host)
  }

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)

    NavigationUI.setupWithNavController(bottom_bar, controller)
  }

  override fun onOptionsItemSelected(item: MenuItem): Boolean {
    val result = super.onOptionsItemSelected(item)
    return NavigationUI.onNavDestinationSelected(item, controller) || result
  }

  override fun onSupportNavigateUp(): Boolean {
    return controller.navigateUp()
  }
}
